src/main/java/com/brightpath/learnify/controller/NotesController.java

```diff
@@ -126,10 +126,17 @@ public ResponseEntity<Void> deleteNote(UUID noteId) {
     }
 
     @Override
-    public ResponseEntity<List<NoteSummaryDto>> listNotes(@Nullable String name, @Nullable String ownerId, @Nullable ResourceAccessTypeDto accessType, @Nullable UUID workspaceId) {
+    public ResponseEntity<List<NoteSummaryDto>> listNotes(
+            @Nullable String name,
+            @Nullable String ownerId,
+            @Nullable ResourceAccessTypeDto accessType,
+            @Nullable UUID workspaceId,
+            @Nullable Float averageRating
+    ) {
         PermissionLevel permissionLevel = dtoMapper.fromResourceAccessTypeDto(accessType);
         String userId = userIdentityService.getCurrentUserId();
-        List<Note> notes = notesService.searchNotes(userId, workspaceId, ownerId, name, permissionLevel);
+        float averageRatingValue = averageRating == null ? 0 : averageRating;
```

Comment:
dont we want use some -1 and handle in on the front side?

Reply:
I think the `0` is such a value. Because rating can be in range 1-5. So filtering by average rating greater then 0 applies to every resource